util/src/main/java/io/grpc/util/AdvancedTlsX509KeyManager.java

```diff
@@ -39,20 +37,21 @@
 
 /**
  * AdvancedTlsX509KeyManager is an {@code X509ExtendedKeyManager} that allows users to configure
- * advanced TLS features, such as private key and certificate chain reloading, etc.
+ * advanced TLS features, such as private key and certificate chain reloading.
+ * We expect only one of {@link AdvancedTlsX509KeyManager#updateIdentityCredentials},
+ * {@link AdvancedTlsX509KeyManager#updateIdentityCredentialsFromFile
+ * (File, File, long, TimeUnit, ScheduledExecutorService)},
+ * {@link AdvancedTlsX509KeyManager#updateIdentityCredentialsFromFile
+ * (File, File, long, TimeUnit, ScheduledExecutorService)} methods to be called after instantiation
+ * of this class.
  */
-@ExperimentalApi("https://github.com/grpc/grpc-java/issues/8024")
 public final class AdvancedTlsX509KeyManager extends X509ExtendedKeyManager {
   private static final Logger log = Logger.getLogger(AdvancedTlsX509KeyManager.class.getName());
-
-  // The credential information sent to peers to prove our identity.
+  // Minimum allowed period for refreshing files with credential information.
+  public final static int MINIMUM_REFRESH_PERIOD = 1 ;
```

Comment:
nit: Please use `Duration` here.

Reply:
They use 'period' across the class here, even in public API like `updateIdentityCredentialsFromFile(File keyFile, File certFile, long period, TimeUnit unit...` so IMHO it's better fits existing style

Reply:
I'll defer to Eric here, but the general Google Java style is to not use primitives to represent time.

Reply:
gRPC code pretty much never uses Duration because we support Android versions that don't have it, and we don't require API-level desugaring. If a specific package doesn't support Android, then it could potentially use it, but it is easier to not use it anywhere. (Streams are probably the one case that are used some when able and not elsewhere. Not to prefer streams, but the author found it valuable.)